conseil-common/src/main/scala/tech/cryptonomic/conseil/common/bitcoin/BitcoinPersistence.scala

```diff
@@ -0,0 +1,131 @@
+    *
+    * @param block JSON_RPC block
+    * @param transactions JSON_RPC block's transactions
+    */
+  def createBlock(
+      block: Block,
+      transactions: List[Transaction]
+  ): DBIOAction[Unit, NoStream, Effect.Write] =
+    DBIO.seq(
+      Tables.Blocks += block.convertTo[Tables.BlocksRow],
+      Tables.Transactions ++= transactions.map(_.convertTo[Tables.TransactionsRow]),
+      Tables.Inputs ++= transactions.map(_.vin).flatten.map(_.convertTo[Tables.InputsRow]),
+      Tables.Outputs ++= transactions.map(_.vout).flatten.map(_.convertTo[Tables.OutputsRow])
+    )
+}
+
+object BitcoinPersistence {
+
+  /**
+    * Create [[cats.Resource]] with [[BitcoinPersistence]].
+    */
+  def resource[F[_]: Concurrent]: Resource[F, BitcoinPersistence[F]] =
+    for {
+      client <- Resource.liftF(
+        Concurrent[F].delay(new BitcoinPersistence[F])
+      )
+    } yield client
+
+  /**
+    * Convert form [[Block]] to [[Tables.BlocksRow]]
+    */
+  implicit val blockToBlocksRow: Conversion[Id, Block, Tables.BlocksRow] = new Conversion[Id, Block, Tables.BlocksRow] {
+    override def convert(from: Block) =
+      Tables.BlocksRow(
+        hash = from.hash,
+        size = from.size,
+        strippedSize = from.strippedsize,
+        weight = from.weight,
+        height = from.height,
+        version = from.version,
+        versionHex = from.versionHex,
+        merkleRoot = from.merkleroot,
+        nonce = from.nonce,
+        bits = from.bits,
+        difficulty = from.difficulty,
+        chainWork = from.chainwork,
+        nTx = from.nTx,
+        previousBlockHash = from.previousblockhash,
+        nextBlockHash = from.nextblockhash,
+        medianTime = new Timestamp(from.mediantime),
+        time = new Timestamp(from.time)
+      )
+  }
+
+  /**
+    * Convert form [[Transaction]] to [[Tables.TransactionsRow]]
+    */
+  implicit val transactionToTransactionsRow: Conversion[Id, Transaction, Tables.TransactionsRow] =
+    new Conversion[Id, Transaction, Tables.TransactionsRow] {
+      override def convert(from: Transaction) =
+        Tables.TransactionsRow(
+          txid = from.txid,
+          blockhash = from.blockhash,
+          hash = from.hash,
+          hex = from.hex,
+          size = from.size,
+          vsize = from.vsize,
+          weight = from.weight,
+          version = from.version,
+          lockTime = new Timestamp(from.locktime),
+          blockTime = new Timestamp(from.blocktime),
+          time = new Timestamp(from.time)
+        )
+    }
+
+  /**
+    * Convert form [[TransactionInput]] to [[Tables.InputsRow]]
+    */
+  implicit val inputToInputsRow: Conversion[Id, TransactionInput, Tables.InputsRow] =
+    new Conversion[Id, TransactionInput, Tables.InputsRow] {
+      override def convert(from: TransactionInput) =
+        Tables.InputsRow(
+          txid = from.txid.get, // TODO: get rid of `get`
```

Comment:
what about this note?

Reply:
It's related to bitcoin client JSON response, I have to reconsider it

Reply:
fixed